rpcserver.go

```diff
@@ -2073,6 +2081,82 @@ func (r *rpcServer) OpenChannelSync(ctx context.Context,
 	}
 }
 
+// BatchOpenChannel attempts to open multiple single-funded channels in a
+// single transaction in an atomic way. This means either all channel open
+// requests succeed at once or all attempts are aborted if any of them fail.
+// This is the safer variant of using PSBTs to manually fund a batch of
+// channels through the OpenChannel RPC.
+func (r *rpcServer) BatchOpenChannel(ctx context.Context,
+	in *lnrpc.BatchOpenChannelRequest) (*lnrpc.BatchOpenChannelResponse,
+	error) {
+
+	if err := r.canOpenChannel(); err != nil {
+		return nil, err
+	}
+
+	// We need the wallet kit server to do the heavy lifting on the PSBT
+	// part. Doesn't make sense to re-implement all of that in the main
+	// server again.
+	var walletKitServer walletrpc.WalletKitServer
+	for _, subServer := range r.subServers {
+		if subServer.Name() == walletrpc.SubServerName {
+			walletKitServer = subServer.(walletrpc.WalletKitServer)
+		}
+	}
+	if walletKitServer == nil {
+		return nil, fmt.Errorf("batch channel open is only possible " +
+			"if walletrpc subserver is active")
+	}
+
+	rpcsLog.Debugf("[batchopenchannel] request to open batch of %d "+
+		"channels", len(in.Channels))
+
+	// Make sure there is at least one channel to open. We could say we want
+	// at least two channels for a batch. But maybe it's nice if developers
+	// can use the same API for a single channel as well as a batch of
+	// channels.
```

Comment:
The existing `OpenChannel` API is quite bloated (has a number of deprecated fields) and is a bit hard to use since it's a streaming response RPC. So I think it's nice to have a more up-to-date alternative.